src/atomic.rs

```diff
@@ -516,8 +516,7 @@ impl<T> Atomic<T> {
     /// });
     /// ```
     pub fn fetch_and<'scope>(&self, val: usize, ord: Ordering, _: &'scope Scope) -> Ptr<'scope, T> {
-        validate_tag::<T>(val);
-        Ptr::from_data(self.data.fetch_and(val, ord))
+        Ptr::from_data(self.data.fetch_and(val | !tag_bits::<T>(), ord))
     }
 
     /// Bitwise "or" with the current tag.
```

Comment:
Please mention in the documentation of all three functions that the tags are truncated to a valid size.